framework/tlvf/src/include/tlvf/CmduMessage.h

```diff
@@ -28,6 +29,16 @@ class CmduMessage {
     template <class T, class = typename std::enable_if<std::is_base_of<BaseClass, T>::value>::type>
     std::shared_ptr<T> addClass()
     {
+        
+        // do not allow to use addClass method if the last tlv is not fully initialized        
+        if(m_class_vector.size() != 0){
+            auto last_tlv = m_class_vector.back();
+            if(!(*last_tlv).isPostInitSucceeded()){
+            TLVF_LOG(ERROR) << "TLV post init failed";
+            return nullptr;
+            }
+        }
```

Comment:
This is full of formatting errors. Did you run clang-format?

Instead of checking m_class_vector.size(), it would be more natural to do:

```
auto last_tlv = m_class_vector.back();
if (last_tlv) {
    TLVF_LOG(ERROR) << "TLV post init failed";
    return nullptr;
}
```